app/code/Magento/UrlRewrite/Block/Catalog/Category/Tree.php

```diff
@@ -162,7 +170,7 @@ protected function _getNodesArray($node)
             'children_count' => (int)$node->getChildrenCount(),
             'is_active' => (bool)$node->getIsActive(),
             // Scrub names for raw js output
-            'name' => $this->escapeHtml($node->getName()),
+            'name' => htmlspecialchars($this->escapeHtml($node->getName()), ENT_COMPAT, 'UTF-8'),
```

Comment:
Instead of `htmlspecialchars` and `$this->escapeHtml` use `$this->escapeHtmlAttr`

Reply:
I have tried using both functions. But didn't turn out to work well as it doesn't escape the string in JSON. the JSON string forms something like **"Test "Cat" 'Cat 1'"**, due to which the string breaks.

Reply:
@vinayshah You are right. I have better solution for both cases. In app/code/Magento/UrlRewrite/view/adminhtml/templates/categories.phtml change from `data-mage-init` attribute to `x-magento-init` script.

Reply:
@omiroshnichenko thanks for this, will try this and if it works will send a PR.

Reply:
Hello @omiroshnichenko 
I am facing this issue in passing the build could you please let me know where am I missing something.
> 1) Magento\UrlRewrite\Block\Catalog\Category\TreeTest::testGetTreeArrayApostropheReplaced
Failed asserting that ''Category 6'' does not contain "'".
/home/travis/build/vinayshah/magento2/dev/tests/integration/testsuite/Magento/UrlRewrite/Block/Catalog/Category/TreeTest.php:60

if escaping single characters with this 
`$this->escapeHtmlAttr($node->getName(), true)`, Then the `_.unescape()` doesn't escape "'", as ASCII code for that is different.